src/storage/v3/shard_rsm.cpp

```diff
@@ -902,24 +944,73 @@ msgs::ReadResponses ShardRsm::HandleRead(msgs::ExpandOneRequest &&req) {
                     req.transaction_id.logical_id);
       break;
     }
-
     if (!req.filters.empty()) {
       // NOTE - DbAccessor might get removed in the future.
-      auto dba = DbAccessor{&acc};
       const bool eval = FilterOnVertex(dba, src_vertex_acc_opt.value(), req.filters, expr::identifier_node_symbol);
       if (!eval) {
         continue;
       }
     }
-    auto result = GetExpandOneResult(acc, src_vertex, req, maybe_filter_based_on_edge_uniquness, edge_filler,
-                                     shard_->GetSchema(shard_->PrimaryLabel()));
 
-    if (!result) {
+    vertex_accessors.emplace_back(src_vertex_acc_opt.value());
+  }
+
+  if (!req.order_by.empty()) {
+    // Can we do differently to avoid this? We need OrderByElements but currently it returns vector<Element>, so this
+    // workaround is here to avoid more duplication later
+    auto sorted_vertices = OrderByVertices(acc, dba, vertex_accessors, req.order_by);
```

Comment:
How can `sorted_vertices` be local variable, shouldn't you iterate over ordered src vertices, if there is order by?

Reply:
This is indeed a local variable that I used to update `vertex_accessors`. The "trick" was done since the return type of `OrderByElements` (now `OrderByVertices`) differed from what we exactly need to keep the code of `HandleRead(msgs::ExpandOneRequest &)`relatively compact and identical for the most part regardless of whether we need ordering or not.
I will look at the return type and see if there is an easy way to avoid this extra transformation. If not, I will at least rename the variable to clarify it's only local and that this is not a mistake.